base/compiler/abstractinterpretation.jl

```diff
@@ -963,8 +988,8 @@ function is_const_prop_profitable_arg(@nospecialize(arg))
     isa(arg, PartialOpaque) && return true
     isa(arg, Const) || return true
     val = arg.val
-    # don't consider mutable values or Strings useful constants
-    return isa(val, Symbol) || isa(val, Type) || (!isa(val, String) && !ismutable(val))
+    # don't consider mutable values useful constants
+    return isa(val, Symbol) || isa(val, Type) || !ismutable(val)
```

Comment:
This is a significant change, so it should land separately (with nanosoldier run)

Reply:
Because `!ismutable(val::String) === false`, doesn't this change any meaning of the condition but just simplify it?

Reply:
Ah, okay, yep

Reply:
Perhaps at some point when this was written String had been immutable?

Reply:
A long, long time ago... (`String` has always been mutable since I started to use this language)